src/controller/CHIPDeviceController.cpp

```diff
@@ -1689,19 +1717,31 @@ void DeviceCommissioner::OnDeviceConnectedFn(void * context, Messaging::Exchange
         return;
     }
 
-    if (commissioner->mDeviceBeingCommissioned == nullptr ||
-        commissioner->mDeviceBeingCommissioned->GetDeviceId() != sessionHandle->GetPeer().GetNodeId())
+    if (commissioner->mCommissioningDelegate == nullptr)
     {
-        // Not the device we are trying to commission.
         return;
```

Comment:
This seems like a logic change: before we would proceed even if `mCommissioningDelegate` was null.  Why the change?

Reply:
Before the changes, we will do nothing if `mCommissioningDelegate` is null. so I moved the check of `mCommissioningDelegate` to here.  If `mCommissioningDelegate` is null, the function will directly return after checking the nodeId.